Main/src/main/webapp/WEB-INF/web.xml

```diff
@@ -20,4 +25,17 @@
         <url-pattern>/api/*</url-pattern>
     </servlet-mapping>
 
+    <filter>
+        <filter-name>HttpServletFilter</filter-name>
+        <filter-class>com.main.acad.servlet.HttpServletFilter</filter-class>
+        <init-param>
+            <param-name>encoding</param-name>
+            <param-value>UTF-8</param-value>
+        </init-param>
+    </filter>
+    <filter-mapping>
+        <filter-name>HttpServletFilter</filter-name>
+        <url-pattern>*.html</url-pattern>
```

Comment:
Your mapping doesn't corresponds to servlet's mapping, that means that your filter will work only for request for html file, but will not work for any ajax request /api/*. Keep in mind that you receive your java articles not through the html files, rather through a string as a response to the ajax request like /api/subchapters